models/remoteworkitem.go

```diff
@@ -0,0 +1,96 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const (
+	ProviderGithub string = "github"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
```

Comment:
@aslakknutsen ,
I was under an impression that derived types would need their own flattening method - but I guess the same flattening method should work across all providers ( as long as they *output* json ). ?

Should I go ahead with a single `Flatten` method belonging to `RemoteWorkItem` and call it inside `MapRemote` ?